docs/reference/benchmarks.py

```diff
@@ -0,0 +1,178 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for generating the table in benchmarks.md."""
+
+import collections
+import multiprocessing
+import os
+import re
+import subprocess
+import sys
+import tarfile
+import zipfile
+
+from common import benchmark_utils
+from common import fuzzer_utils
+from common import utils
+from common import oss_fuzz
+
+BUILD_ARCHIVE_EXTENSION = '.tar.gz'
+LEN_BUILD_ARCHIVE_EXTENSION = len(BUILD_ARCHIVE_EXTENSION)
+COVERAGE_BUILD_PREFIX = 'coverage-build-'
+LEN_COVERAGE_BUILD_PREFIX = len(COVERAGE_BUILD_PREFIX)
+
+GUARDS_REGEX = re.compile(r'INFO:.*\((?P<num_guards>\d+) guards\).*')
+
+ONE_MB = 1024**2
+
+BENCHMARK_INFO_FIELDS = ['benchmark', 'target', 'dict', 'seeds', 'guards', 'MB']
+BenchmarkInfo = collections.namedtuple('BenchmarkInfo', BENCHMARK_INFO_FIELDS)
+
+
+def get_benchmark_infos(builds_dir):
+    """Get BenchmarkInfo for each benchmark that has a build in
+    builds_dir."""
+    build_paths = [
+        os.path.join(builds_dir, path)
+        for path in os.listdir(builds_dir)
+        if path.endswith(BUILD_ARCHIVE_EXTENSION)
+    ]
+    pool = multiprocessing.Pool()
+    return pool.map(get_benchmark_info, build_paths)
+
+
+def get_real_benchmark_name(benchmark):
+    """The method we use to infer benchmark names from coverage builds
+    doesn't quite work because the project name is used in OSS-Fuzz
+    builds instead. This function figures out the actual benchmark based on
+    the project name."""
+    benchmarks_dir = os.path.join(utils.ROOT_DIR, 'benchmarks')
+    real_benchmarks = [
+        real_benchmark for real_benchmark in os.listdir(benchmarks_dir)
+        if os.path.isdir(os.path.join(benchmarks_dir, real_benchmark))
+    ]
+    if benchmark in real_benchmarks:
+        return benchmark
+    for real_benchmark in real_benchmarks:
+        if not benchmark_utils.is_oss_fuzz(real_benchmark):
+            continue
+        config = oss_fuzz.get_config(real_benchmark)
+        if config['project'] == benchmark:
+            return real_benchmark
+    return None
+
+
+def count_oss_fuzz_seeds(fuzz_target_path):
+    """Count the number of seeds in the OSS-Fuzz seed archive for
+    |fuzze_target_path|."""
+    zip_file_name = fuzz_target_path + '_seed_corpus.zip'
+    if not os.path.exists(zip_file_name):
+        return 0
+    seeds = 0
+    with zipfile.ZipFile(zip_file_name) as zip_file:
+        for info in zip_file.infolist():
+            if info.filename.endswith('/'):
+                continue
+            seeds += 1
+    return seeds
+
+
+def count_standard_seeds(seeds_path):
+    """Count the number of seeds for a standard benchmark."""
```

Comment:
consider something like `len([p for p in Path(seed_path).glob('**/*') if p.is_file()])`

(`from pathlib import Path`)